docker/bookkeeper/entrypoint.sh

```diff
@@ -74,6 +73,28 @@ create_bookie_dirs() {
   done
 }
 
+# Create a Bookie ID if this is a newly added bookkeeper pod
+# or read the Bookie ID if it already exists
+set_bookieid() {
+  IFS=',' read -ra directories <<< $1
+  BK_ID_DIR="${directories[0]}"
+  BK_ID_FILE="${BK_ID_DIR}/id"
+  if [ `find ${BK_ID_FILE} | wc -l` -gt 0 ]; then
+    echo "Found BookieID"
+    BK_bookieId=`cat ${BK_ID_FILE}`
+  else
+    if [ `find $BK_journalDirectory $BK_ledgerDirectories $BK_indexDirectories -type f 2> /dev/null | wc -l` -gt 0 ]; then
+      HOST="$(echo -e `hostname -A` | sed -e 's/[[:space:]]*$//')"
+      BK_bookieId="${HOST}:${BOOKIE_PORT}"
+    else
+      HOST="`hostname -s`"
+      BK_bookieId="${HOST}:${RANDOM}"
+    fi
+  fi
+  echo "BookieID = $BK_bookieId"
+  sed -i "s|.*bookieId=.*\$|bookieId=${BK_bookieId}|" ${BK_HOME}/conf/bk_server.conf
```

Comment:
I suppose this works with the base image we are using? I'm asking because this command does not work on OS X, but it doesn't necessarily have to.

Reply:
Yes @fpj , I have verified that it works through my testing. 
Also we were already using something similar in the same file itself https://github.com/pravega/pravega/blob/ca2ba631c74122d38b894d084523f6c3d379f05c/docker/bookkeeper/entrypoint.sh#L113-L115